File: src/main/java/ch/dafo90/swissqrbillgenerator/model/Field.java
package ch.dafo90.swissqrbillgenerator.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Field {

    private String name;
    private String defaultMap;
    private String label;
    private boolean required;
    private String requiredText;
    private FieldType type;
    private String defaultValue;
    private List<String> options;

}

File: frontend/src/app/types/field.ts
import {FieldType} from "./field-type";

export interface Field {

    name: string;
    defaultMap: string;
    label: string;
    required: boolean;
    requiredText: string | null;
    type: FieldType;
    defaultValue: string | null;
    options: string[] | null;
    allowStaticValue: boolean;

}
